File: ruby/p002.rb
#
# Solution to Project Euler problem 2
# by Alex Reichert
#
# https://github.com/reichert621/Project-Euler-solutions
#

def compute
  sum = 0
  x = 1
  y = 2

  while (x < 4000000)
    sum += x if x.even?

    x, y = y, x + y
  end

  sum
end

File: javascript/p002.js
'use strict';

module.exports = compute;

function compute() {
  let sum = 0;
  let x = 1;
  let y = 2;

  while (x < 4000000) {
    if (x % 2 === 0) sum += x;

    let tmp = x;
    x = y;
    y = tmp + y;
  }

  return sum;
}
